projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/interactions/cartesian-explorer-context-menu/cartesian-explorer-context-menu.component.ts

```diff
@@ -38,16 +43,21 @@ export class CartesianExplorerContextMenuComponent<TData> {
 
   public constructor(
     @Inject(POPOVER_DATA) data: CartesianSelectedData<TData>,
-    private readonly cartesainExplorerNavigationService: CartesainExplorerNavigationService
+    private readonly cartesainExplorerNavigationService: CartesainExplorerNavigationService,
+    private readonly timeRangeService: TimeRangeService
   ) {
     this.selectionData = data;
   }
 
   public menuSelectHandler = (_menu: ContextMenu): void => {
-    this.cartesainExplorerNavigationService.navigateToExplorer(
-      this.selectionData.timeRange.startTime,
-      this.selectionData.timeRange.endTime
-    );
+    this.timeRangeService.setFixedRange(this.selectionData.timeRange.startTime, this.selectionData.timeRange.endTime);
```

Comment:
This shouldn't need to be set here, the time range is passed to the navigateToExplorer method and set as query params on https://github.com/hypertrace/hypertrace-ui/pull/1149/files#diff-707e0c6ee1d2d1e2a7f5365ff8a899673d103e05e8e8c619b5f3bbc1de0382f9R19

Reply:
@aaron-steinfeld I think there are two use cases.
1. You want to select a fraction of time and apply the same on the dashboard page. (You choose the menu option `Apply Time Range`)
2. You want to apply the time range and navigate to explorer dashboard. (you use `Explore` menu option).

So, we need to set the time range for case `1` here. For `2`, you are correct that the `navigateToExplorer` method handles setting the time range via `queryParams`, as we are not calling `navigateToExplorer` for menu option `1`.

Reply:
Right, I think this belies the problem in the design though - every context menu item calls the same handler. We want this called in case 1 above, but not in case 2 where the navigation params for time are set directly.

At a minimum, we can change this method to switch on the calling item, but I think a significantly better solution and no real extra work would be to add a handler to the `ContextMenu` interface and change line 20 to `            (click)="menu.onClick()"`.

Reply:
So, to add the respective handler for each item within the `menu` object definition itself, right?
Like the `ContextMenu` interface will be having one more property called `onClick` which will be the reference of the function to be executed whenever the respective item is clicked.

If that's the case, I think it makes sense and will make it generic enough.

Reply:
Ahh, thanks for the clarification, I might have put up a reference like `this.someMethod` as you said. Will keep this in mind.

Reply:
@aaron-steinfeld when we navigate to explorer without setFixedRange, the global time range filter is not changing. see in the below screenshot it is showing 15 mins, on the other hand, URL shows with correct parameters
![Screenshot 2022-01-06 at 9 25 15 AM](https://user-images.githubusercontent.com/86998924/148326057-eaafbe14-244f-4489-bfcd-ea2beab358eb.png)

Reply:
Hmm, that may be an outstanding bug. Guessing the time range service doesn't keep track of the URL after it initializes, which it should be. Do you mind filing it as a separate issue? For now, we can just separate the handlers and add the extra explicit set in the explorer nav handler to get this moving, then come back to fix the url handling and remove.